File: frontend/src/microservices/advertising/objects/AdvertisementOwnerActionRequest.ts
// 本文件定义 advertising 模块的 `AdvertisementOwnerActionRequest`，用于广告所有者发起的提交/暂停等动作并提供 JSON 编解码。

export type AdvertisementOwnerActionRequest = {
  advertisementId: string
  ownerManagerId: string
  ownerType: string
}

export const advertisementOwnerActionRequestFromJson = (json: string): AdvertisementOwnerActionRequest =>
  JSON.parse(json) as AdvertisementOwnerActionRequest

export const advertisementOwnerActionRequestToJson = (value: AdvertisementOwnerActionRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/AdvertisementOwnerActionRequest.scala
// 本文件定义 advertising 模块的 AdvertisementOwnerActionRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AdvertisementOwnerActionRequest(
    advertisementId: String,
    ownerManagerId: String,
    ownerType: String
)

object AdvertisementOwnerActionRequest:
  given sourceEncoder: Encoder[AdvertisementOwnerActionRequest] = deriveEncoder
  given sourceDecoder: Decoder[AdvertisementOwnerActionRequest] = deriveDecoder

